tests/std/tests/P0448R4_spanstream/test.cpp

```diff
@@ -0,0 +1,1027 @@
+        CharT output_buffer[30];
+        basic_ospanstream<CharT> os{span<CharT>{output_buffer}};
+
+        assert(os.good());
+        assert(!os.fail());
+        assert(!os.bad());
+        os << 10 << 20 << 30;
+        assert(os.good());
+        assert(!os.fail());
+        assert(!os.bad());
+
+        const auto expected = "102030"sv;
+        assert(os.span().size() == 6);
+        assert(equal(begin(os.span()), end(os.span()), begin(expected), end(expected)));
+        assert(os.span().data() == output_buffer);
+    }
+
+    { // write to stream with overflow
+        CharT output_buffer[30];
+        basic_ospanstream<CharT> os{span<CharT>{output_buffer}};
+
+        os << 10 << 20 << 30;
+        assert(os.good());
+        assert(!os.fail());
+        assert(!os.bad());
+        if constexpr (is_same_v<CharT, char>) {
+            os << "hello world and a long string with more than 30 chars";
+        } else {
+            os << L"hello world and a long string with more than 30 chars";
+        }
+        assert(!os.good());
+        assert(os.fail());
+        assert(os.bad());
+
+        const auto expected = "102030hello world and a long s"sv;
+        assert(os.span().size() == size(output_buffer));
+        assert(equal(begin(output_buffer), end(output_buffer), begin(expected), end(expected)));
+    }
+}
+
+template <class CharT>
+void test_spanstream() {
+    using test_buf = test_buf<basic_spanbuf<CharT>>;
+    { // construction
+        CharT buffer[10];
+        basic_spanstream<CharT> span_constructed{span<CharT>{buffer}};
+        assert(span_constructed.span().data() == buffer);
+        assert(static_cast<test_buf*>(span_constructed.rdbuf())->eback() == buffer);
+        assert(static_cast<test_buf*>(span_constructed.rdbuf())->gptr() == buffer);
+        assert(static_cast<test_buf*>(span_constructed.rdbuf())->egptr() == end(buffer));
+        assert(static_cast<test_buf*>(span_constructed.rdbuf())->pbase() == buffer);
+        assert(static_cast<test_buf*>(span_constructed.rdbuf())->pptr() == buffer);
+        assert(static_cast<test_buf*>(span_constructed.rdbuf())->epptr() == end(buffer));
+
+        basic_spanstream<CharT> span_mode_constructed{span<CharT>{buffer}, ios_base::in};
+        assert(span_mode_constructed.span().data() == buffer);
+        assert(static_cast<test_buf*>(span_mode_constructed.rdbuf())->eback() == buffer);
+        assert(static_cast<test_buf*>(span_mode_constructed.rdbuf())->gptr() == buffer);
+        assert(static_cast<test_buf*>(span_mode_constructed.rdbuf())->egptr() == end(buffer));
+        assert(static_cast<test_buf*>(span_mode_constructed.rdbuf())->pbase() == nullptr);
+        assert(static_cast<test_buf*>(span_mode_constructed.rdbuf())->pptr() == nullptr);
+        assert(static_cast<test_buf*>(span_mode_constructed.rdbuf())->epptr() == nullptr);
+    }
+
+    { // span
+        CharT buffer[10];
+        basic_ospanstream<CharT> is{span<CharT>{buffer}};

```

Comment:
This is `test_spanstream()`, but a `basic_ospanstream` is here (also named `is`). Below there's a `basic_ispanstream<CharT> is`. Will need to check for any other occurrences.